### PR TITLE
#162633453 Setup react testing environment

### PR DESCRIPTION
**Pull request by mycok**
**What does this PR do?**

- Setup a react testing environment

**Description of Task to be completed?**
- After cloning the repo, `cd` to the project root folder and run `npm install` to install dependencies.

**How should this be manually tested?**

- Use `npm test` or `yarn test` to run the tests.
- You could also run `yarn/ npm test --coverage` to run tests with coverage.

**Any background context you want to provide?**

- `Incase you get dependency errors, delete the node_modules folder along with the package-lock.json files `
- `remove "jest": "^24.1.0",from package.json`
- `run npm cache clean`
- `run command npm install`
- `then run npm install jest@latest --save-dev`

**What are the relevant pivotal tracker stories?**

[#162633453](https://www.pivotaltracker.com/story/show/162633453)

### CODE REVIEW

File: coverage/clover.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<coverage generated="1549996349688" clover="3.2.0">
+  <project timestamp="1549996349688" name="All files">
+    <metrics statements="1" coveredstatements="1" conditionals="0" coveredconditionals="0" methods="1" coveredmethods="1" elements="2" coveredelements="2" complexity="0" loc="1" ncloc="1" packages="1" files="1" classes="1"/>
+    <file name="App.js" path="/Users/michael.k/Desktop/ah-react/ah-frontend-valkyrie/src/App.js">
```
Comment (jeanjoe) [inline coverage/clover.xml]: This file path is referencing your local file, it may not be useful here

**Review by jeanjoe — CHANGES_REQUESTED** [review]

File: coverage/coverage-final.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{"/Users/michael.k/Desktop/ah-react/ah-frontend-valkyrie/src/App.js": {"path":"/Users/michael.k/Desktop/ah-react/ah-frontend-valkyrie/src/App.js","statementMap":{"0":{"start":{"line":5,"column":4},"end":{"line":9,"column":6}}},"fnMap":{"0":{"name":"(anonymous_0)","decl":{"start":{"line":4,"column":2},"end":{"line":4,"column":3}},"loc":{"start":{"line":4,"column":11},"end":{"line":10,"column":3}},"line":4}},"branchMap":{},"s":{"0":1},"f":{"0":1},"b":{},"_coverageSchema":"43e27e138ebf9cfc5966b082cf9a028302ed4184","hash":"ea37af48b1d70c7f98d619556e4af9e44102ef20"}
```
Comment (jeanjoe) [inline coverage/coverage-final.json]: Please remove this local file path

**Review by jeanjoe — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -23,14 +26,20 @@
...
     "html-webpack-plugin": "^3.2.0",
+    "jest": "^24.1.0",
     "webpack": "^4.29.3",
     "webpack-cli": "^3.2.3",
-    "webpack-dev-server": "^3.1.14"
+    "webpack-dev-server": "^3.1.14",
+    "react-test-renderer": "^16.8.1"
   }
-}
+}
```
Comment (mubstimor) [inline package.json:56]: missing new line here

File: tests/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    // Indicates whether each individual test should be reported during the run
+    verbose: false,
+
+    // Add jest support for various file extensions including css and image files
+    moduleNameMapper: {
+        "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2)$": "<rootDir>/src/fileMock.js",
+        "\\.(css|scss)$": "<rootDir>/src/styleMock.js"
+    }
+};
```
Comment (mubstimor) [inline tests/jest.config.js]: same missing line here

**Comment by araaliFarooq** [comment]
LGTM

**Review by jeanjoe — APPROVED** [review]

**Comment by jeanjoe** [comment]
LGTM

**Comment by Qadriyah** [comment]
LGTM

**Comment by NabunyaLilian** [comment]
Good work @mycok but please resolve the violations.

**Comment by araaliFarooq** [comment]
> * You could also run `yarn/ npm test --coverage` to run tests with coverage.

when running tests  with coverage using **yarn** , command is ` yarn test --coverage` and if using **npm**  the comand is `npm test -- --coverage` .

**Pull request merged**